### ISSUE TITLE
changing repeat_rate forgets keys that are held down

### ISSUE THREAD

**Issue by JasonWoof** [swaywm/sway#6776]
Thanks for sway! It's awesome :)

I have super+number keybinds set up to change workspaces. So I hold down super and press number keys to flip around to find something.

After many months, I lost my patience with vscode's bug where they get the key repeat wrong (because they're using an old version of electron, which uses an old version of chrome.) so I wrote a little script to automatically change the key repeat when focus switches to vscode (and back.)

```bash
#!/usr/bin/env bash

pressesPerSecond=70

prevRate="$pressesPerSecond"
while read x; do
    if [ "$x" = Code ]; then
        rate="$((1000 / $pressesPerSecond))";
    else
        rate="$pressesPerSecond";
    fi
    if [ "$rate" != "$prevRate" ]; then
        swaymsg "input type:keyboard repeat_rate $rate"
        prevRate="$rate"
    fi
done < <(swaymsg -t subscribe -m '[ "window" ]' | jq -r --unbuffered .container.app_id)
```

It's ridiculously cool that this is possible and so easy in sway.

This bug is about what happens when I try to use both of these features at once.

With that script active, I hold down super and press number keys. As soon as I switch to- or from vscode it forgets that I'm holding down super, and starts typing numbers into whatever program has focus. Same issue with my keys to switch focus between windows in the same workspace.

I'm testing sway 1.6.1 on nixos. My apologies if this has been fixed already.

**Comment by JasonWoof** [comment]
P.S. My use case is a little weird, but I assume this would also affect people who want a different keyboard layout for different windows: https://github.com/artemsen/swaykbdd

**Comment by RPigott** [comment]
Wouldn't this be a client bug? Might need a WAYLAND_DEBUG log to tell but it sounds like the client just isn't disarming its timers properly.

**Comment by JasonWoof** [comment]
This has nothing to do with clients. I'm pretty sure I could reproduce this with no client windows open.

I'm pressing super+2 which is bound like so:

`bindsym $mod+2 workspace n2`

It usually works, unless I've changed the repeat_rate recently in which case sway does two things wrong:

1. It fails to change workspaces
2. It sends a keypress event (for the "2" key) to a client

**Comment by RPigott** [comment]
Can you provide a sway debug log?

**Comment by RPigott** [comment]
Oh I understand what you're saying now. You don't mean that the held keys are repeating, you are pressing the 2 key again and expect it to trigger the $mod+2 keybind, but it doesn't. Is that right?

**Comment by JasonWoof** [comment]
Yeah, this has nothing to do with repeating.

The problem is that when I set the key repeat rate with swaymsg while holding down super, sway acts like I'm not holding down super anymore.

You should be able to reproduce this by running this:

`while sleep 1; do swaymsg "input type:keyboard repeat_rate $rate"; done`

Then hold down your modifier key and try to use your sway keybinds that use that mod key.

I suspect the same thing happens when changing keyboard layout, which would be very problematic for people who use a different keyboard layout for different windows.

**Comment by RPigott** [comment]
Yes I can reproduce this. Probably can be fixed whenever input config is revamped for #6654.

**Comment by zDEFz** [comment]
I can reproduce this without even changing anything.
It happens seemingly random with `scratchpad show` where the first time I show scratchpad it works and the second time it forgot META is held  down.

I got a warning in wev, though

Unable to mmap keymap: Invalid argument[14:     wl_keyboard] repeat_info: rate: 25 keys/sec; delay: 600 ms
